### ISSUE TITLE
Unsafe handling of TryToIdentifySingleton in ClusterSingletonProxy

### ISSUE THREAD

**Issue by danischroeter** [akka/akka#24311]
`ClusterSingletonProxy` internally schedules `TryToIdentifySingleton` to self.
Under some circumstances this leads to 
```
java.io.NotSerializableException: No configured serialization-bindings for class [akka.cluster.singleton.ClusterSingletonProxy$TryToIdentifySingleton$]
    at akka.serialization.Serialization.serializerFor(Serialization.scala:236)
    at akka.serialization.Serialization.findSerializerFor(Serialization.scala:212)
    at akka.remote.MessageSerializer$.serialize(MessageSerializer.scala:44)
    at akka.remote.EndpointWriter.$anonfun$serializeMessage$1(Endpoint.scala:889)
    at scala.util.DynamicVariable.withValue(DynamicVariable.scala:58)
    at akka.remote.EndpointWriter.serializeMessage(Endpoint.scala:889)
    at akka.remote.EndpointWriter.writeSend(Endpoint.scala:780)
    at akka.remote.EndpointWriter$$anonfun$4.applyOrElse(Endpoint.scala:755)
    at akka.actor.Actor.aroundReceive(Actor.scala:513)
    at akka.actor.Actor.aroundReceive$(Actor.scala:511)
    at akka.remote.EndpointActor.aroundReceive(Endpoint.scala:446)
    at akka.actor.ActorCell.receiveMessage(ActorCell.scala:527)
    at akka.actor.ActorCell.invoke_aroundBody0(ActorCell.scala:496)
    at akka.actor.ActorCell$AjcClosure1.run(ActorCell.scala:1)
    at org.aspectj.runtime.reflect.JoinPointImpl.proceed(JoinPointImpl.java:149)
    at akka.kamon.instrumentation.ActorMonitors$$anon$1.$anonfun$processMessage$1(ActorMonitor.scala:63)
    at kamon.trace.Tracer$.withContext(TracerModule.scala:58)
    at akka.kamon.instrumentation.ActorMonitors$$anon$1.processMessage(ActorMonitor.scala:63)
    at akka.kamon.instrumentation.ActorCellInstrumentation.aroundBehaviourInvoke(ActorInstrumentation.scala:45)
    at akka.actor.ActorCell.invoke(ActorCell.scala:489)
    at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
    at akka.dispatch.Mailbox.run(Mailbox.scala:224)
    at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
    at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
    at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
    at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
    at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

The reason this happens is not as first guessable `serialize-messages = on` is set (which would also fail. The reason is that the receive checks for `case ClusterSingletonProxy.TryToIdentifySingleton if identifyTimer.isDefined ⇒` where identifyTimer is the Option of the cancellable for the message. Since it is possible that we cancel but the message already made it to the actors mailbox. This means that once `TryToIdentifySingleton` with cancelled `identifyTimer` happens this matches `case msg: Any ⇒` so `ClusterSingletonProxy` will try to send `TryToIdentifySingleton` to the singleton. If the default java serialization is disabled this means a `java.io.NotSerializableException: No configured serialization-bindings` and the proxy will restart or with default serialization the singleton will have an unmatched/unexpected message.

It looks to me that `if identifyTimer.isDefined` should be removed...? Opinions?

**Comment by johanandren** [comment]
That message is never intended to be sent over remoting, so a simple workaround would be to not `serialize-messages = on`. (See #24273 for a related recent discussion)

**Comment by danischroeter** [comment]
@johanandren as mentioned the problem here is not that the message is serialized - the problem is that `ClusterSingletonProxy` should not forward but handle its own message.

**Comment by johanandren** [comment]
Oh, I'm sorry, you did describe it very clearly, not sure how I missed that. Yeah. We should indeed make sure that message does not end up in the forward-catchall in some way.

Up for PR:ing a fix?

**Comment by danischroeter** [comment]
@johanandren no problem. PR:ing: sure thing. But would like to have your opinions on the approach for the fix.

**Comment by johanandren** [comment]
Gut feeling is just put the if inside of the case instead of as a guard, if `identifTimer` is `None` that should mean that we already managed to identify the singleton and then we can ignore it.

Would be good if @patriknw could verify that I didn't get it wrong though :)

**Comment by ktoso** [comment]
Looking into this

**Comment by ktoso** [comment]
Thanks for reporting @danischroeter ! Will be fixed in 2.5.10
